common/page.go

```diff
@@ -1039,8 +1116,74 @@ func (p *Page) Workers() []api.Worker {
 	return workers
 }
 
+func (p *Page) onConsoleAPICalled(event *cdpruntime.EventConsoleAPICalled) {
+	// If there are no handlers for EventConsoleAPICalled, return
+	p.eventHandlersMu.RLock()
+	if _, ok := p.eventHandlers[eventPageConsoleAPICalled]; !ok {
+		p.eventHandlersMu.RUnlock()
+		return
+	}
+	p.eventHandlersMu.RUnlock()
+
+	m, err := p.consoleMssgFromConsoleEvent(event)
+	if err != nil {
+		p.logger.Errorf("Page:onConsoleAPICalled", "building console message: %v", err)
+		return
+	}
+
+	p.eventHandlersMu.RLock()
+	defer p.eventHandlersMu.RUnlock()
+	for _, h := range p.eventHandlers[eventPageConsoleAPICalled] {
+		// Use TaskQueue in order to synchronize handlers execution in the event loop,
+		// as it is not thread safe and events are processed from a background goroutine
+		p.tq.Queue(func() error {
+			if err := h(m); err != nil {
+				return fmt.Errorf("executing onConsoleAPICalled handler: %w", err)
+			}
+			return nil
+		})
+	}
+}
+
+func (p *Page) consoleMssgFromConsoleEvent(e *cdpruntime.EventConsoleAPICalled) (*ConsoleMessage, error) {
```

Comment:
```suggestion
func (p *Page) consoleMsgFromConsoleEvent(e *cdpruntime.EventConsoleAPICalled) (*ConsoleMessage, error) {
```

Reply:
Force pushed with the change in https://github.com/grafana/xk6-browser/pull/1006/commits/12f6f4a6038e20c250fac2498f3cc37ecdf25b6a. See https://github.com/grafana/xk6-browser/blob/f5abd5d865238b52bc046a18ed082dcdfe4ed138/common/page.go#L1148.